OasisDEX/oasis-borrow
# Create 1inch proxy, and route 1inch calls from front-end through it.

# [Create 1inch proxy, and route 1inch calls from front-end through it.](https://app.shortcut.com/oazo-apps/story/10212/create-1inch-proxy-and-route-1inch-calls-from-front-end-through-it)

<please insert a shortcut link above>
  
## Changes 👷‍♀️
  <Please add short list of changes>
Implemented a new api route /api/exchange to proxy calls to the 1inch api with added authentication header.
Auth header is using a static key from .env which is put into the header under the `auth-key` prop.
Refactored swap helpers and exchange views to use a new api proxy.
  
## How to test 🧪
  Open eth multiply position which is using this proxy API to 1inch and follow up to complete the flow and open position succesfully. Check network tab in browser dev tools that it's using the proxy endpoint /api/exchange instead of 1inch api directly.
